### ISSUE TITLE
OPN plugin can cause slow loading of the WordPress plugins page

### ISSUE THREAD

**Issue by cagross** [cagross/outdated-plugin-notifier#1]
Currently the OPN plugin retrieves the 'last updated' date for each installed plugin by making HTTP calls to the WordPress plugin API--one HTTP call after another, in sequence.  This can cause the WordPress plugins page to load relatively slowly--the more plugins are installed, the longer the page will take to load.

The HTTP calls to the plugin API is done via the [plugins_api()](https://developer.wordpress.org/reference/functions/plugins_api/) function.  More research is necessary to confirm whether it is possible to optimize these HTTP calls, e.g. ensure asynchronous calls, or fetch all plugin data with a single HTTP call.

**Comment by cagross** [comment]
This issue was fixed in commit 9c7faf1 (i.e. version 1.0.2).  That commit ensured HTTP calls to the WordPress plugins API were done via AJAX, rather than during PHP execution (via the `plugins_api()` function), resulting in much faster page load speeds.

One additional step *could* be to ensure the plugin code first checks WordPress transients for 'last updated' date of each plugin.  Then, the plugin would fetch data from the WordPress plugins API only if the information was not in the transient cache.  This *could* further improve page load speed.  But the speed already appears to be <1 second, so any improvements would be relatively small.  I've added this to the 'to do' list, and maybe it will become an official GitHub issue in the future.
